python/name-service/name_service/__init__.py

```diff
@@ -38,18 +50,32 @@ def get_year():
         r = requests.get(YEAR_ENDPOINT)
         return int(r.text)
 
+# shared resource service name
+myResource = Resource.create({"service.name": "name-python"})
 
-trace.set_tracer_provider(TracerProvider(
-    resource=Resource.create({SERVICE_NAME: "name-python"})
-))
+# tracing pipeline
+trace.set_tracer_provider(TracerProvider(resource=myResource))
 tracer = trace.get_tracer_provider().get_tracer(__name__)
+trace_exporter = OTLPSpanExporter(
+        headers=(("x-honeycomb-team", os.environ.get("HONEYCOMB_API_KEY")),),
+        endpoint=os.environ.get("HONEYCOMB_API_ENDPOINT",
+                                "https://api.honeycomb.io")
+    )
+trace.get_tracer_provider().add_span_processor(BatchSpanProcessor(trace_exporter))
 
-trace.get_tracer_provider().add_span_processor(
-    BatchSpanProcessor(OTLPSpanExporter(
+# logging pipeline
+logger_provider = LoggerProvider(resource=myResource)
+set_logger_provider(logger_provider)
```

Comment:
I think we should move the `set_logger_provider` to after we've finished configuring it.

Reply:
I've re-arranged, can you take a look and let me know if it matches what you were saying?